### ISSUE TITLE
registry secret sometimes does not populate available options

### ISSUE THREAD

**Issue by slickwarren** [rancher/dashboard#5132]
**Setup**
- Rancher version: v2.6.4-rc2
- Browser type & version:chrome

**Describe the bug**

when an existing registry secret is available, sometimes they do not show up at all in the list, giving the user only an option to create a new one. 

**To Reproduce**

deploy a cluster using a registry with an http secret
once deployed, you can either go to the cluster -> view config option and refresh the page a few times
or go to create a new cluster and try to use the same secret

**Result**
secret(s) don't display for the registry

**screenshots**
<img width="393" alt="Screen Shot 2022-02-15 at 2 15 50 PM" src="https://user-images.githubusercontent.com/16691014/154159256-a16098f9-5d37-4249-a312-0741451af281.png">

**Comment by nwmac** [comment]
@cnotv Can you see if you can reproduce this bug

**Comment by cnotv** [comment]
For the records, this is the mentioned dropdown:

![Screenshot from 2022-03-08 14-03-38](https://user-images.githubusercontent.com/5009481/157243553-884e867f-847f-4911-b76a-5649b66b4bf2.png)

**Comment by cnotv** [comment]
@slickwarren while creating a cluster based on a registry with HTTP basic auth, the provision task gets interrupted and shows the following log, which seems quite cryptic:

![Screenshot from 2022-03-08 17-13-59](https://user-images.githubusercontent.com/5009481/157278540-01d9bcbd-2131-43d7-bca9-d05499e714ac.png)

![Screenshot from 2022-03-08 17-12-59](https://user-images.githubusercontent.com/5009481/157278381-e9858dd1-4fa4-46d4-a391-ea220a192b60.png)

Furthermore if I try to create a new cluster, I can see the generated credentials still there.

![Screenshot from 2022-03-08 16-58-03](https://user-images.githubusercontent.com/5009481/157278911-27946dfb-2e93-49f1-a370-aa9a19f1d4cc.png)

The auth options are also listed in `local`:

![Screenshot from 2022-03-08 17-18-16](https://user-images.githubusercontent.com/5009481/157279588-74193d25-5645-434f-a841-c248c69d290d.png)

**Comment by izaac** [comment]
Rancher version: `v2.6.4-rc12`
I tried a few times with a private registry, in both k3s and rke2 cluster types. 

I am able to see the secret on edit of an existing cluster and in the create new cluster.
I also see multiple secrets listed

Closing, if we are able to get reproducible steps we can reopen or link to a new issue.

![Screen Shot 2022-03-23 at 11 05 22 PM](https://user-images.githubusercontent.com/27659/159853105-65da8d16-8d13-47ec-bd56-e2d2cf127be0.png)
![Screen Shot 2022-03-23 at 11 05 47 PM](https://user-images.githubusercontent.com/27659/159853109-b860380b-2fb2-44c1-8ee5-2a862df294c7.png)
